### ISSUE TITLE
Users with Curator role get system emails after the Curator role has been removed

### ISSUE THREAD

**Issue by philippconzett** [IQSS/dataverse#7235]
One of the curators in one of the sub-dataverses in our repository was removed as a curator some weeks ago. But it seems the system is still trying to send emails about submitted dataset etc. to her - we get undeliverable-messages from the email server of her former employer.

**Comment by djbrooke** [comment]
Hi @philippconzett, is it possible that the person still has a role somewhere within the system? As a first step I'd maybe suggest running the query for notifications from the useful queries doc (https://docs.google.com/document/d/1-Y_iUduSxdDNeK1yiGUxe7t-Md7Fy965jp4o4m1XEoE/edit#) and seeing where this person's name shows up: 

select date_trunc('year', senddate), type, count(type)
from usernotification
--where type = 7
group by type, date_trunc('year', senddate)
order by type, date_trunc('year', senddate);

There may be another query that would be more appropriate for this (such as a query that just shows the roles for a person) but it doesn't look like there are any in the document. 

cc: @jggautier to see if you have any additional thoughts.

**Comment by jggautier** [comment]
Are any of these emails about datasets submitted to the sub-dataverse that she used to have a curator role on, or could they be notifications about datasets (or other types of notifications) from other sub-dataverses? I wasn't sure from your post, @philippconzett. If you're not sure, then I agree getting info about the users notifications would help.

The results of the database query @djbrooke found would look like this:

<img width="848" alt="Screen Shot 2020-08-28 at 1 13 05 PM" src="https://user-images.githubusercontent.com/18374574/91596066-8fe4e200-e932-11ea-9e46-021460f3f860.png">

It might also helpful to narrow the results using the ID of the user (you could find the user ID in the superuser dashboard or with other database queries) and get more info about those numbers in the "objectid" column (which could be a dataset, dataverse or file) like what type of object it is and which dataverse it's in, using a query like:

```
select usernotification.emailed, usernotification.objectid, usernotification.senddate, usernotification.type, usernotification.user_id, dvobject.dtype, dvobject.owner_id, dataverse.name
from usernotification
join dvobject on dvobject.id = usernotification.objectid
join dataverse on dataverse.id = owner_id
where usernotification.user_id = ####
order by usernotification.senddate desc
```

Which gives you:

<img width="964" alt="Screen Shot 2020-08-28 at 1 31 33 PM" src="https://user-images.githubusercontent.com/18374574/91596793-ebaf6b00-e932-11ea-985b-03154fcf849c.png">

_(couldn't figure out what those numbers in the "type" column mean)_

In addition to database queries, there's an API endpoint for [getting info about a particular user's notifications](http://guides.dataverse.org/en/5.0/api/native-api.html#get-all-notifications-by-user) but I don't think the results of those API calls would be very helpful here and you'd need the user's API token.

There's an API endpoint for [listing a user's role assignments](http://guides.dataverse.org/en/5.0/api/native-api.html#list-role-assignments-of-a-role-assignee), but it's a "admin"-level API endpoint and I don't know how to use those. Not sure how helpful the results of those API calls would be.

Lastly, I tested this scenario using Demo Dataverse and three different accounts, where account-1 had its curator role taken from a sub-dataverse by account-2, then account-3 submitted a dataset for review to that sub-dataverse. Account-1 didn't get a notification about the submitted dataset.

Hope that's helpful!

**Comment by philippconzett** [comment]
Thanks, @djbrooke and @jggautier, for useful advice. I have managed to create a query that provides an overview of all (non-deleted?) user notifications sent to a specified user. I had to modify Julian's SQL script somewhat [1]. The resulting table shows that the curator I was talking about indeed received notifications on the day we got the latest undelivarable email notice (2020-08-27):

![image](https://user-images.githubusercontent.com/21955790/91654103-8a100d80-eaa6-11ea-9034-f97cfe1eb3b6.png)

In the user overview in the dashboard I noticed that the mentioned curator was listed with an admin role. This was strange, because as I mentioned earlier I had removed this user from both the curator and the admin user group for that sub-dataverse. But I now discovered that her user was assigned the admin role directly under Permissions. As a general rule, we don't grant access permission to individual users, only to user groups. I don't know why her user got assigned the admin role directly in the Permissions settings and not through the admin user group.

Sorry for occupying your time with this silly mistake!

-----
[1] Combining the table usernotification with the table dvoject has to go through the table authenticateduser. I used the SQL script below. Note that the prefix "public." might not be necessary in your application / database.

SELECT public.dataverse.name,
	public.usernotification.id,
	public.usernotification.emailed,
	public.usernotification.objectid,
	public.usernotification.readnotification,
	public.usernotification.senddate,
	public.usernotification.type,
	public.usernotification.user_id,
	public.usernotification.requestor_id
FROM (public.usernotification
	INNER JOIN public.authenticateduser
		ON public.usernotification.user_id = public.authenticateduser.id)
	INNER JOIN (public.dvobject INNER JOIN public.dataverse
		ON public.dvobject.owner_id = public.dataverse.id)
		ON public.authenticateduser.id = public.dvobject.releaseuser_id
WHERE (((public.usernotification.user_id)=XXX))
ORDER BY public.usernotification.senddate DESC;
